overlord/snapstate/snapstate.go

```diff
@@ -2565,6 +2569,22 @@ func RemoveMany(st *state.State, names []string) ([]string, []*state.TaskSet, er
 	return removed, tasksets, nil
 }
 
+func validateSnapNames(names []string) error {
+	var invalidNames []string
+
+	for _, name := range names {
+		if err := snap.ValidateInstanceName(name); err != nil {
+			invalidNames = append(invalidNames, name)
+		}
+	}
+
+	if len(invalidNames) > 0 {
+		return fmt.Errorf("found invalid snap names: %v", strings.Join(invalidNames, ", "))
```

Comment:
(nitpick) most of our error strings start with `cannot`, so maybe something like: `cannot remove invalid snap names: $v` ?